### PR TITLE
Print the complete error when fails

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Clean the rest like #349

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., Usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm

File: support/kind/kind.go [hunk 1]
```diff
@@ -174,14 +174,11 @@ func (k *Cluster) Create(ctx context.Context, args ...string) (string, error) {
...
-		// Print the output data as well so that it can be useful to debug cluster bringup failures
-		var data []byte
-		b := bytes.NewBuffer(data)
-		_, err := io.Copy(b, p.Out())
+		outBytes, err := io.ReadAll(p.Out())
 		if err != nil {
 			log.ErrorS(err, "failed to read data from the kind create process output due to an error")
 		}
-		return "", fmt.Errorf("failed to create kind cluster: %s : %s: %s", p.Err(), p.Result(), b.String())
+		return "", fmt.Errorf("failed to create kind %q cluster: %s: %s: %s", k.name, p.Err(), p.Result(), string(outBytes))
```
Comment (harshanarayana) [inline support/kind/kind.go:181]:
```suggestion
		return "", fmt.Errorf("failed to create kind cluster %q: %s: %s: %s", k.name, p.Err(), p.Result(), string(outBytes))
```

might be more readable ?

File: support/kwok/kwok.go [hunk 2]
```diff
@@ -143,7 +143,7 @@ func (k *Cluster) Create(ctx context.Context, args ...string) (string, error) {
 		if err != nil {
 			klog.ErrorS(err, "failed to read data from the kwok create process output due to an error")
 		}
-		return "", fmt.Errorf("failed to create kwok cluster: %s: %s: %s", p.Err(), p.Result(), string(outBytes))
+		return "", fmt.Errorf("failed to create kwok %q cluster: %s: %s: %s", k.name, p.Err(), p.Result(), string(outBytes))
```
Comment (harshanarayana) [inline support/kwok/kwok.go:146]: Same as above.

File: support/kind/kind.go [hunk 3]
```diff
@@ -236,7 +233,11 @@ func (k *Cluster) Destroy(ctx context.Context) error {
 
 	p := utils.RunCommand(fmt.Sprintf(`%s delete cluster --name %s`, k.path, k.name))
 	if p.Err() != nil {
-		return fmt.Errorf("kind: delete cluster %v failed: %s: %s", k.name, p.Err(), p.Result())
+		outBytes, err := io.ReadAll(p.Out())
+		if err != nil {
+			log.ErrorS(err, "failed to read data from the kind delete process output due to an error")
+		}
+		return fmt.Errorf("kind: delete cluster %q cluster: %s: %s: %s", k.name, p.Err(), p.Result(), string(outBytes))
```
Comment (harshanarayana) [inline support/kind/kind.go:240]:
```suggestion
		return fmt.Errorf("kind: delete cluster %q: %s: %s: %s", k.name, p.Err(), p.Result(), string(outBytes))
```

File: support/kwok/kwok.go [hunk 4]
```diff
@@ -175,7 +175,11 @@ func (k *Cluster) Destroy(ctx context.Context) error {
 
 	p := utils.RunCommand(fmt.Sprintf(`%s delete cluster --name %s`, k.path, k.name))
 	if p.Err() != nil {
-		return fmt.Errorf("kwok: delete cluster failed: %s: %s", p.Err(), p.Result())
+		outBytes, err := io.ReadAll(p.Out())
+		if err != nil {
+			klog.ErrorS(err, "failed to read data from the kwok delete process output due to an error")
+		}
+		return fmt.Errorf("kwok: delete cluster %q cluster: %s: %s: %s", k.name, p.Err(), p.Result(), string(outBytes))
```
Comment (harshanarayana) [inline support/kwok/kwok.go:182]:
```suggestion
		return fmt.Errorf("kwok: delete cluster %q: %s: %s: %s", k.name, p.Err(), p.Result(), string(outBytes))
```

**Review by harshanarayana — CHANGES_REQUESTED** [review]

**Comment by wzshiming** [comment]
Done

**Review by vladimirvivien — APPROVED** [review]
/lgtm

**Pull request merged**